File: src/EasyPrinter/Models/ScanSettings.cs
namespace EasyPrinter.Models
{
    /// <summary>
    /// Настройки сканирования
    /// </summary>
    public class ScanSettings
    {
        /// <summary>
        /// Разрешение сканирования (DPI)
        /// </summary>
        public ScanResolution Resolution { get; set; } = ScanResolution.Dpi300;

        /// <summary>
        /// Формат сохранения
        /// </summary>
        public ScanFormat Format { get; set; } = ScanFormat.PDF;

        /// <summary>
        /// Папка для сохранения
        /// </summary>
        public string OutputFolder { get; set; } = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments);

        /// <summary>
        /// Имя файла (без расширения)
        /// </summary>
        public string FileName { get; set; } = $"Скан_{DateTime.Now:yyyy-MM-dd_HH-mm-ss}";

        /// <summary>
        /// Источник сканирования
        /// </summary>
        public ScanSource Source { get; set; } = ScanSource.Flatbed;

        /// <summary>
        /// Настройки изображения
        /// </summary>
        public ImageAdjustments ImageAdjustments { get; set; } = new();

        /// <summary>
        /// Получить полный путь к файлу
        /// </summary>
        public string GetFullPath()
        {
            var extension = Format switch
            {
                ScanFormat.PDF => ".pdf",
                ScanFormat.JPEG => ".jpg",
                ScanFormat.PNG => ".png",
                ScanFormat.TIFF => ".tiff",
                _ => ".pdf"
            };

            return Path.Combine(OutputFolder, FileName + extension);
        }
    }

    public enum ScanResolution
    {
        Dpi150 = 150,
        Dpi300 = 300,
        Dpi600 = 600,
        Dpi1200 = 1200
    }

    public enum ScanFormat
    {
        PDF,
        JPEG,
        PNG,
        TIFF
    }

    public enum ScanSource
    {
        Flatbed,    // Стекло сканера
        ADF         // Автоподатчик документов
    }
}

File: easyprinter/models/scan_settings.py
"""
Настройки сканирования
"""

from dataclasses import dataclass, field
from enum import Enum
from pathlib import Path
from datetime import datetime
import os

from .image_adjustments import ImageAdjustments


class ScanResolution(Enum):
    """Разрешение сканирования (DPI)"""
    DPI_150 = 150
    DPI_300 = 300
    DPI_600 = 600
    DPI_1200 = 1200


class ScanFormat(Enum):
    """Формат сохранения"""
    PDF = "pdf"
    JPEG = "jpg"
    PNG = "png"
    TIFF = "tiff"


class ScanSource(Enum):
    """Источник сканирования"""
    FLATBED = "flatbed"     # Стекло сканера
    ADF = "adf"             # Автоподатчик документов


@dataclass
class ScanSettings:
    """Настройки сканирования"""

    # Разрешение сканирования (DPI)
    resolution: ScanResolution = ScanResolution.DPI_300

    # Формат сохранения
    format: ScanFormat = ScanFormat.PDF

    # Папка для сохранения
    output_folder: str = field(default_factory=lambda: str(Path.home() / "Documents"))

    # Имя файла (без расширения)
    file_name: str = field(default_factory=lambda: f"Скан_{datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}")

    # Источник сканирования
    source: ScanSource = ScanSource.FLATBED

    # Настройки изображения
    image_adjustments: ImageAdjustments = field(default_factory=ImageAdjustments)

    def get_full_path(self) -> str:
        """Получить полный путь к файлу"""
        extension = self.format.value
        return os.path.join(self.output_folder, f"{self.file_name}.{extension}")
